packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -0,0 +1,186 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../common/implementation/Lockable.sol";
+import "../../common/implementation/FixedPoint.sol";
+import "../../common/implementation/Testable.sol";
+
+import "../../oracle/interfaces/StoreInterface.sol";
+import "../../oracle/interfaces/FinderInterface.sol";
+import "../../oracle/implementation/Constants.sol";
+
+import "../funding-rate-store/interfaces/FundingRateStoreInterface.sol";
+
+
+/**
+ * @title FundingRateApplier contract.
+ * @notice Provides funding rate payment functionality for the Perpetual contract.
+ */
+
+contract FundingRateApplier is Testable, Lockable {
+    using SafeMath for int256;
+    using SafeMath for uint256;
+    using FixedPoint for FixedPoint.Unsigned;
+    using SafeERC20 for IERC20;
+
+    /****************************************
+     * FUNDING RATE APPLIER DATA STRUCTURES *
+     ****************************************/
+
+    // Points to financial product-related contracts like the funding rate store.
+    FinderInterface public fpFinder;
```

Comment:
I did this because we previously kept non-oracle related contracts outside of the Finder, so I wasn't sure if we wanted to re-use the "oracle" finder for these non-oracle contracts

Reply:
Seems reasonable for now. We can think more about it down once everything is piped together.